### PR TITLE
article-store sample Deployment resource

### PR DESCRIPTION
**Pull request by giorgiosironi**
Part of https://github.com/libero/publisher/issues/360

### CODE REVIEW

File: helm/publisher/.helmignore [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Patterns to ignore when building packages.
```
Comment (giorgiosironi) [inline helm/publisher/.helmignore]: what is this garbage? Generated by Helm, would remove most of it

Comment (giorgiosironi) [reply helm/publisher/.helmignore]: No one should be building packages locally anyway.

File: helm/publisher/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v2
+name: publisher
+description: Libero Publisher vanilla flavor
+type: application
+version: 0.1.0
+appVersion: latest
```
Comment (giorgiosironi) [inline helm/publisher/Chart.yaml:6]: this is meaningless at the moment - sometimes used to select the Docker image tag but not the case here

File: helm/publisher/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{{/* vim: set filetype=mustache: */}}
```
Comment (giorgiosironi) [inline helm/publisher/templates/_helpers.tpl]: deleted all the boilerplate I could find here

File: helm/publisher/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "publisher.name" -}}
+{{- .Chart.Name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
```
Comment (giorgiosironi) [inline helm/publisher/templates/_helpers.tpl]: given the level of reusability of this chart I'm tempted to strip the 63-character checks and if not to support names such as `publisher--my-staging-environment-that-really-really-needs-a-long-name` in favor of `publisher--my-staging-environment-that-really-needs-a-long-name`

File: helm/publisher/templates/scholarly-articles-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        {{- include "publisher.selectorLabels" . | nindent 8 }}
+        app.kubernetes.io/component: scholarly-articles
+    spec:
+      containers:
+        - name: app
+          image: "{{ .Values.images.articleStore.repository }}:latest"
+          imagePullPolicy: {{ .Values.images.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 8080
```
Comment (giorgiosironi) [inline helm/publisher/templates/scholarly-articles-deployment.yaml]: would prefer 80 for this, but the container image dictates here

Comment (giorgiosironi) [reply helm/publisher/templates/scholarly-articles-deployment.yaml]: (technically not needed yet until there is a `Service`)

File: helm/publisher/templates/scholarly-articles-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}--scholarly-articles
```
Comment (thewilkybarkid) [inline helm/publisher/templates/scholarly-articles-deployment.yaml]: Might be best to just call it `article-store` for now. Need to experiment with how to distinguish between two article stores in the hypermedia world.

Comment (giorgiosironi) [reply helm/publisher/templates/scholarly-articles-deployment.yaml]: No opinion, will leave it generic. The important part is to distinguish it from other Deployments like a frontend container, a search or a gateway service, etc.

**Pull request merged**